llvm/lib/Analysis/MemoryProfileInfo.cpp

```diff
@@ -271,6 +300,27 @@ static void saveFilteredNewMIBNodes(std::vector<Metadata *> &NewMIBNodes,
     }
   };
 
+  if (MostlyCold) {
```

Comment:
Can you add a short comment describing why we can return early if MostlyCold is true?

Reply:
done